File: wlr/OGUR/OGUR/Items/ItemClass.cs
using System.Collections.Generic;
namespace OGUR.Items
{
    public class ItemClass
    {
        public const int NULL = 0;
        public const int Melee_Weapon = 1;
        public const int Ranged_Weapon = 2;
        public const int Ranged_Ammo = 3;
        public const int Ring = 4;
        public const int Leggings = 5; 
        public const int Torso_Garb = 6;
        public const int Feet = 7;
        public const int Head_Gear = 8;
        public const int Gloves = 9;
        public const int Shield = 10;
        public const int LAST = 11;

        public static readonly List<int> Values = new List<int>()
        {
            NULL,
            Melee_Weapon,
            Ranged_Weapon, 
            Ranged_Ammo,
            Ring,
            Leggings, 
            Torso_Garb,
            Feet,
            Head_Gear,
            Gloves,
            Shield,
            LAST
        };

        public static readonly Dictionary<int, string> Names = new Dictionary<int, string>()
        {
            {NULL,"NULL"},
            {Melee_Weapon,"Melee Weapon"},
            {Ranged_Weapon,"Ranged Weapon"}, 
            {Ranged_Ammo,"Ranged Ammo"},
            {Ring,"Ring"},
            {Leggings,"Leggings"}, 
            {Torso_Garb,"Torso Garb"},
            {Feet,"Feet"},
            {Head_Gear,"Head Gear"},
            {Gloves,"Gloves"},
            {Shield,"Shield"},
            {LAST,"LAST"}
        };
    }
}

File: code/aigilas-ported/src/com/aigilas/items/ItemClass.java
package com.aigilas.items;import com.xna.wrapper.*;import java.util.*;    public class ItemClass
    {
        public static final int NULL = 0;
        public static final int Melee_Weapon = 1;
        public static final int Ranged_Weapon = 2;
        public static final int Ranged_Ammo = 3;
        public static final int Ring = 4;
        public static final int Leggings = 5; 
        public static final int Torso_Garb = 6;
        public static final int Feet = 7;
        public static final int Head_Gear = 8;
        public static final int Gloves = 9;
        public static final int Shield = 10;
        public static final int LAST = 11;

        public static int[] Values =
        {
            NULL,
            Melee_Weapon,
            Ranged_Weapon, 
            Ranged_Ammo,
            Ring,
            Leggings, 
            Torso_Garb,
            Feet,
            Head_Gear,
            Gloves,
            Shield,
            LAST
        };

        public static HashMap<Integer, String> Names = new HashMap<Integer, String>();
        static{
             Names.put(NULL,"NULL");
             Names.put(Melee_Weapon,"Melee Weapon");
             Names.put(Ranged_Weapon,"Ranged Weapon"); 
             Names.put(Ranged_Ammo,"Ranged Ammo");
             Names.put(Ring,"Ring");
             Names.put(Leggings,"Leggings"); 
             Names.put(Torso_Garb,"Torso Garb");
             Names.put(Feet,"Feet");
             Names.put(Head_Gear,"Head Gear");
             Names.put(Gloves,"Gloves");
             Names.put(Shield,"Shield");
             Names.put(LAST,"LAST");
        };
    }
